### ISSUE TITLE
Issue Regarding Future Warnings

### ISSUE THREAD

**Issue by troymyname** [ray-project/ray#26424]
Hi, I am currently at Ray 1.13.0, and am getting a lot warnings regarding deprecation of code from Pandas, SciPy, etc. Could these be taken care of? Please review below.

2022-07-09 06:04:11,456	INFO services.py:1470 -- View the Ray dashboard at http://127.0.0.1:8265
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(RolloutWorker pid=13041) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13041)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13041) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13041)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13040) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13040)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13040) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13040)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13044) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13044)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13044) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13044)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13042) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13042)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13042) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13042)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13046) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13046)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13046) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13046)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13048) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13048)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13048) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13048)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13041) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13041)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13040) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13040)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13044) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13044)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13042) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13042)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13046) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13046)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13048) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13048)   from pandas import Int64Index as NumericIndex
2022-07-09 06:04:17,804	INFO trainable.py:588 -- Restored on 192.168.0.180 from checkpoint: /home/tonmoy/PycharmProjects/zeus/__trade/modeSavePath/PPO_DAILY/PPO_fc762_00000_0_2022-06-30_17-38-31/checkpoint_001680/checkpoint-1680
2022-07-09 06:04:17,804	INFO trainable.py:597 -- Current state after restoring: {'_iteration': 1680, '_timesteps_total': None, '_time_total': 710563.6827175617, '_episodes_total': 8124}
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (4.0.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(RolloutWorker pid=13317) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13317)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13317) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13317)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13315) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.bagging module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13315)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13315) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.ensemble.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.ensemble. Anything that cannot be imported from sklearn.ensemble is now part of the private API.
(RolloutWorker pid=13315)   warnings.warn(message, FutureWarning)
(RolloutWorker pid=13317) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13317)   from pandas import Int64Index as NumericIndex
(RolloutWorker pid=13315) /home/tonmoy/PycharmProjects/zeus/venv/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
(RolloutWorker pid=13315)   from pandas import Int64Index as NumericIndex

**Comment by kouroshHakha** [comment]
Hi, can you narrow down which import statement is causing this stack trace? Also it may be that your packages versions are not those that ray 1.13.0 requires.

**Comment by gjoliver** [comment]
can you create a minimal script (doesn't have to do any real work) that triggers these warnings?
we'd love to do some tests on our side. especially about the sklearn warnings.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
